### PR TITLE
Use an Enum like class for annotation permissions

### PR DESCRIPTION
**Pull request by jon-betts**
Yep, looks like good permissions coverage in the functional tests.

### CODE REVIEW

File: h/security/permissions.py [hunk 1]
```diff
@@ -1,4 +1,14 @@
 class Permission:
+    class Annotation:
+        ADMIN = "annotation:admin"  # Is this used anywhere?
```
Comment (seanh) [inline h/security/permissions.py:3]: I don't think this is used anywhere. There's no concept of "administrating an annotation" that I can recall. I think we can remove this?

File: h/security/permissions.py [hunk 2]
```diff
@@ -1,4 +1,14 @@
...
+        READ = "annotation:read"
+        WRITE = "annotation:write"  # Is this granted anywhere?
```
Comment (seanh) [inline h/security/permissions.py:5]: I think delete this. It doesn't seem to make any sense, we already have `"annotation:create"` and `"annotation:update"` below

File: h/services/oauth_validator.py [hunk 3]
```diff
@@ -6,12 +6,13 @@
...
 from h import models
 from h.models.auth_client import GrantType as AuthClientGrantType
+from h.security.permissions import Permission
 from h.services.oauth_provider import ACCESS_TOKEN_PREFIX, REFRESH_TOKEN_PREFIX
 from h.util.db import lru_cache_in_transaction
 from h.util.uri import render_url_template
 
 AUTHZ_CODE_TTL = datetime.timedelta(minutes=10)
-DEFAULT_SCOPES = ["annotation:read", "annotation:write"]
+DEFAULT_SCOPES = [Permission.Annotation.READ, Permission.Annotation.WRITE]
```
Comment (seanh) [inline h/services/oauth_validator.py:15]: I think we should leave this as `["annotation:read", "annotation:write"]`. I don't think these are Pyramid permissions. They're set as the default scopes for OAuth access tokens. But I don't see them actually being used for anything

**Review by seanh — CHANGES_REQUESTED** [review]
Looks good except I don't think we should change the OAuth scopes. As far as I can tell those are unrelated to Pyramid permissions

File: h/services/oauth_validator.py (hunk 3 above)
Comment (jon-betts) [reply h/services/oauth_validator.py:15]: Oh I see what happened here. 

I first changed all the strings like `"read"` into `"group:read"` etc. I picked colon so as not to clash with lots of our route definitions like `group.read`. These must have already been like this and just got caught up in the replacement.

I didn't think there was anything in the code base which followed this naming convention.

File: h/security/permissions.py (hunk 1 above)
Comment (jon-betts) [reply h/security/permissions.py:3]: Yep that's dealt with here: https://github.com/hypothesis/h/pull/6784

File: h/security/permissions.py (hunk 2 above)
Comment (jon-betts) [reply h/security/permissions.py:5]: Similarly this is done in: https://github.com/hypothesis/h/pull/6784

**Pull request merged**